### PR TITLE
Maintain current Pivot selection when saving on the Profiles page

### PR DESCRIPTION
**Pull request by zadjii-msft**
This PR Makes sure that after you save the settings, we stay on the same part of the profiles pivot. We do this by having a singleton `ProfilesNavigationState`, a bit like the color scheme one in #8799. Hence why this PR is targeting the other.

## PR Checklist
* [x] I work here
* [x] Tested manually
* [x] Fixes the first point in #8769

### CODE REVIEW

File: src/cascadia/TerminalSettingsEditor/MainPage.cpp [hunk 1]
```diff
@@ -226,7 +235,12 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         {
             auto profileVM{ _viewModelForProfile(_settingsClone.ProfileDefaults()) };
             profileVM.IsBaseLayer(true);
-            contentFrame().Navigate(xaml_typename<Editor::Profiles>(), winrt::make<ProfilePageNavigationState>(profileVM, _settingsClone.GlobalSettings().ColorSchemes(), *this));
+
+            // Update the profiles navigation state
+            _profilesNavState.Profile(profileVM);
+            _profilesNavState.Schemes(_settingsClone.GlobalSettings().ColorSchemes());
```
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/MainPage.cpp:241]: Do you even need to update the reference to `Schemes` here? Shouldn't we just update the profile?

File: src/cascadia/TerminalSettingsEditor/MainPage.cpp [hunk 2]
```diff
@@ -240,12 +254,14 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 
     void MainPage::_Navigate(const Editor::ProfileViewModel& profile)
     {
-        auto state{ winrt::make<ProfilePageNavigationState>(profile, _settingsClone.GlobalSettings().ColorSchemes(), *this) };
+        // Update the profiles navigation state
+        _profilesNavState.Profile(profile);
+        _profilesNavState.Schemes(_settingsClone.GlobalSettings().ColorSchemes());
```
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/MainPage.cpp:259]: Same question as above.

**Review by carlos-zamora — APPROVED** [review]
So excited for this to go in! Thanks!

File: src/cascadia/TerminalSettingsEditor/MainPage.cpp (hunk 1 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalSettingsEditor/MainPage.cpp:241]: I honestly don't know. Before, we were taking a new reference to the ColorSchemes on each navigation - I bet we don't anymore

**Review by DHowett — APPROVED** [review]

**Pull request merged**